terrain.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+import argparse
+import base64
+import getpass
+import jwt
+import os
+import os.path
+import requests
+import stat
+import sys
+
+terrain_base_urls = {
+    "prod": "https://de.cyverse.org/terrain",
+    "qa": "https://qa.cyverse.org/terrain",
+}
+
+def terrain_auth_file(environment):
+    """
+    Returns the Terrain auth token file path for the given environment.
+    """
+    return "{0}/.terrain-{1}".format(os.environ["HOME"], environment)
+
+def get_cached_access_token(environment):
+    """
+    Attempts to obtain the access token from $HOME/.terrain-auth-{environment}.
```

Comment:
It looks like `terrain_auth_file` will look for the file at `$HOME/.terrain-{environment}`?
```suggestion
    Attempts to obtain the access token from $HOME/.terrain-{environment}.
```

Reply:
Ah! Thanks for catching these 👍